### PR TITLE
Use "git checkout" instead of "git reset" when checkout git tip

### PR DESCRIPTION
**Pull request by kiennq**
This is a refactored PR from #152 
In this change:
- Change from `git reset --hard` to `git checkout` so we don't accidentally lose commit when trying to test local change not committed.

### CODE REVIEW

**Comment by steckerhalter** [comment]
Thanks for the PR... there's an issue with the build though, can you fix the error? I think it should be ok to merge after the build is succeeding. Cheers

**Comment by kiennq** [comment]
> Thanks for the PR... there's an issue with the build though, can you fix the error? I think it should be ok to merge after the build is succeeding. Cheers

Build passed now

File: quelpa.el [hunk 1]
```diff
@@ -216,13 +216,15 @@ On error return nil."
...
-                             (with-no-warnings (package-tar-file-info)))))))))
+                           (with-no-warnings
+                             (if (help-function-arglist 'package-tar-file-info)
+                                 ;; legacy `package-tar-file-info' requires an arg
+                                 (package-tar-file-info file)
+                               (package-tar-file-info)))))))))
     (pcase desc
       ((pred package-desc-p) desc)
-      ((pred vectorp) (package-desc-from-legacy desc kind)))))
+      ((pred vectorp) (if (fboundp 'package-desc-from-legacy)
```
Comment (alphapapa) [inline quelpa.el]: Please prefer `when` over "one-armed ifs".

File: quelpa.el [hunk 2]
```diff
@@ -1734,9 +1746,8 @@ If t, `quelpa' tries building the stable version of a package."
 (defun quelpa-package-install-file (file)
   "Workaround problem with `package-install-file'.
 `package-install-file' uses `insert-file-contents-literally'
-which causes problems when the file inserted has crlf line
-endings (Windows). So here we replace that with
-`insert-file-contents' for non-tar files."
+which causes problems when the FILE inserted has crlf line endings (Windows).
```
Comment (alphapapa) [inline quelpa.el]: Please run `checkdoc` on your docstring changes.  This should be wrapped to 70 columns.

**Comment by steckerhalter** [comment]
@kiennq Sorry I merged this too early. Can you make the changes @alphapapa suggested in a another PR? Thanks

**Pull request merged**